scripts/svdpatch.py

```diff
@@ -354,22 +355,27 @@ def process_peripheral_regs_array(ptag, rspec, rmod):
     li, ri = spec_ind(rspec)
     for rtag in list(iter_registers(ptag, rspec)):
         rname = rtag.findtext('name')
-        registers.append([rname, rtag, rname[li:len(rname)-ri], int(rtag.findtext('addressOffset'), 0)])
+        registers.append([rtag, rname[li:len(rname)-ri], int(rtag.findtext('addressOffset'), 0)])
     dim = len(registers)
-    if dim > 0:
-        registers = sorted(registers, key=lambda r: r[3])
-        regarray = registers[0]
+    if dim == 0:
+        raise SvdPatchError("{}: registers {} not found"
+                .format(ptag.findtext('name'), rspec))
+    else:
```

Comment:
We don't need the `else` here at all: the `raise` above will return from the function, so the rest of the code can be unindented. Same on the second check later.

Reply:
In my opinion if we get zero size register specification matching, then patch is incorrect, and we should make signal. Maybe just print("Warn: ...") ?

Reply:
I agree the patch would be incorrect - we can raise the error as you do here, but then we don't need to put the rest of the code in `else` because it will only run if the error wasn't raised anyway.

Reply:
Ok. Now I understand what you want.